src/lava/proc/event_data/event_data_loader/aedat_data_loader.py

```diff
@@ -0,0 +1,137 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import typing as ty
+
+from dv import AedatFile
+import numpy as np
+import random
+from operator import itemgetter
+import os
+
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.process.ports.ports import OutPort
+
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.magma.core.model.py.ports import PyOutPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.resources import CPU
+from lava.magma.core.decorator import implements, requires
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+
+
+class AedatDataLoader(AbstractProcess):
+    def __init__(self,
+                 file_path: str,
+                 shape_out: tuple,
+                 seed_sub_sampling: int = None,
+                 **kwargs) -> None:
+        super().__init__(file_path=file_path,
+                         shape_out=shape_out,
+                         seed_sub_sampling=seed_sub_sampling,
+                         **kwargs)
+
+        self._validate_file_path(file_path)
+        self._validate_shape_out(shape_out)
+
+        self.out_port = OutPort(shape=shape_out)
+
+    @staticmethod
+    def _validate_file_path(file_path):
+        # Checking file extension
+        if not file_path[-7:] == ".aedat4":
+            raise ValueError(f"Given file should be an .aedat4 file. "
+                             f"{file_path} given.")
+
+        try:
+            # Checking file size
+            if os.stat(file_path).st_size > 0:
+                return file_path
+        except FileNotFoundError:
+            # Checking file exists
+            raise FileNotFoundError(f"File not found. {file_path} given.")
+
+        return file_path
+
+    @staticmethod
+    def _validate_shape_out(shape_out):
+        if not isinstance(shape_out[0], int):
+            raise ValueError(f"Max number of events should be an integer."
+                             f"{shape_out} given.")
```

Comment:
done, implemented every where type annotations were not used

Reply:
Keep in mind though that the type annotations are just hints, they do not check anything at runtime (see my other comment above). So, wherever you think it is crucial, I would still add a validation that throws an exception. Otherwise, just leave it out.

Reply:
For now if the user inputs an invalid dimension or negative values in the shapes it throws an error. I guess we can keep it like this for now.